lightning/src/events/bump_transaction.rs

```diff
@@ -231,3 +267,486 @@ pub enum BumpTransactionEvent {
+	///    throughout coin selection.
+	/// 3. Enough inputs must be selected/contributed for the resulting transaction (including the
+	///    inputs and outputs noted above) to meet `target_feerate_sat_per_1000_weight`.
+	///
+	/// Implementations must take note that [`Input::witness_weight`] only tracks the weight of the
+	/// input's witness. Some wallets, like Bitcoin Core's, may require providing the full input
+	/// weight. Failing to do so may lead to underestimating fee bumps and delaying block inclusion.
+	///
+	/// The `claim_id` uniquely identifies a claim, and should be used to assign a set of UTXOs to
+	/// the claim, such that they can be re-used within new fee-bumped iterations of the original
+	/// claiming transaction, ensuring that claims don't double spend each other. If a specific
+	/// `claim_id` has never had a transaction associated with it, and all of the available UTXOs
+	/// have already been assigned to other claims, implementations must be willing to double spend
+	/// their UTXOs. The choice of which UTXOs to double spend is left to the implementor, e.g., an
+	/// implementation could choose to re-assign the UTXOs assigned to the lowest value claim.
+	fn select_confirmed_utxos(
+		&self, claim_id: ClaimId, must_spend: Vec<Input>, must_pay_to: Vec<TxOut>,
+		target_feerate_sat_per_1000_weight: u32,
+	) -> Result<CoinSelection, ()>;
+	/// Returns a script to use for change above dust resulting from a successful coin selection
+	/// attempt.
+	fn change_script(&self) -> Result<Script, ()>;
+	/// Signs and provides the full witness for all inputs within the transaction known to the
+	/// trait (i.e., any provided via [`CoinSelectionSource::select_confirmed_utxos`]).
+	fn sign_tx(&self, tx: &mut Transaction) -> Result<(), ()>;
+}
+
+/// An alternative to [`CoinSelectionSource`] that can be implemented and used along [`Wallet`] to
+/// provide a default implementation to [`CoinSelectionSource`].
+pub trait WalletSource {
+	/// Returns all UTXOs, with at least 1 confirmation each, that are available to spend.
+	fn list_confirmed_utxos(&self) -> Result<Vec<Utxo>, ()>;
+	/// Returns a script to use for change above dust resulting from a successful coin selection
+	/// attempt.
+	fn change_script(&self) -> Result<Script, ()>;
+	/// Signs and provides the full witness for all inputs within the transaction known to the
+	/// wallet (i.e., any provided via [`WalletSource::list_confirmed_utxos`]).
+	fn sign_tx(&self, tx: &mut Transaction) -> Result<(), ()>;
+}
+
+/// A wrapper over [`WalletSource`] that implements [`CoinSelection`] by preferring UTXOs that would
+/// avoid conflicting double spends. If not enough UTXOs are available to do so, conflicting double
+/// spends may happen.
+pub struct Wallet<W: Deref> where W::Target: WalletSource {
+	source: W,
+	// TODO: Do we care about cleaning this up once the UTXOs have a confirmed spend? We can do so
```

Comment:
Maybe after some confirmation depth ? Like if there is a shallow reorg of 1-block, we might still keep the UTXO locked?